carml-engine/src/test/java/io/carml/engine/iotests/RmlMapperTest.java

```diff
@@ -194,6 +194,21 @@ void testSettingCustomBaseIri() {
         builder -> builder.baseIri("http://foo.bar/base/"));
   }
 
-  // TODO: PM: add test for rml:reference and rr:template where a value is not found.
+  @Test
+  void testMappingWithNullReference() {
+    testMapping("RmlMapper", "/RmlMapper/nullValues/nullValueReference.rml.ttl",
+        "/RmlMapper/nullValues/nullValueReference.output.ttl");
+  }
 
+  @Test
+  void testMappingWithTemplateWithNullReference() {
+    testMapping("RmlMapper", "/RmlMapper/nullValues/nullValueTemplate.rml.ttl",
+        "/RmlMapper/nullValues/nullValueTemplate.output.ttl");
+  }
+
+  @Test
+  void testMappingWithTemplateWithMultiValuedNullReference() {
+    testMapping("RmlMapper", "/RmlMapper/nullValues/nullValueMultiValuedTemplate.rml.ttl",
+        "/RmlMapper/nullValues/nullValueMultiValuedTemplate.output.ttl");
```

Comment:
`null` value in subject map, no triples are generated at all, correct?

Reply:
correct